core/pkg/artifacts/linker.go

```diff
@@ -72,3 +81,41 @@ func (al *ArtifactLinker) Link() error {
 
 	return err
 }
+
+func (al *ArtifactLinker) resolveOrgEntityName(portfolioEntity string, organization string) (string, error) {
+	// Fetches the org entity of the portfolio entity to
+	// 1. validate the user inputted the correct display org name or org entity name and
+	// 2. return the org entity name so we can use the correct entity name to link the artifact.
+	orgFieldNames, err := GetGraphQLFields(al.Ctx, al.GraphqlClient, "Organization")
+	if err != nil {
+		return "", err
+	}
+	canFetchOrgEntity := slices.Contains(orgFieldNames, "orgEntity")
+	if organization == "" && !canFetchOrgEntity {
+		return "", fmt.Errorf("Fetching Registry artifacts without inputting an organization is unavailable for your server version. Please upgrade your server to 0.50.0 or later.")
+	}
+	if !canFetchOrgEntity {
+		// Use traditional registry path with org entity if server doesn't support it
+		return organization, nil
+	}
+
+	response, err := gql.FetchOrgEntityFromEntity(
+		al.Ctx,
+		al.GraphqlClient,
+		portfolioEntity,
+	)
+	if err != nil {
+		return "", err
+	}
+	if response == nil || response.GetEntity() == nil || response.GetEntity().GetOrganization() == nil || response.GetEntity().GetOrganization().GetOrgEntity() == nil {
+		return "", fmt.Errorf("Unable to find organization for artifact under entity: %s. Please make sure you are using a team entity when linking to the Registry", portfolioEntity)
+	}
+
+	// Validate organization inputted by user
+	orgEntityName := response.Entity.Organization.OrgEntity.Name
+	inputOrgMatchesOrgNameOrOrgEntityName := (organization == orgEntityName || organization == response.Entity.Organization.Name)
+	if organization != "" && !inputOrgMatchesOrgNameOrOrgEntityName {
```

Comment:
This would be more naturally written using two variables:

```go
inputMatchesOrgName := organization == orgEntityName
inputMatchesOrgEntityName := organization == response.Entity.Organization.Name
if organization != "" &&
	!inputMatchesOrgName &&
	!inputMatchesOrgEntityName {
```